File: wrappers/fedimint-py/models/wallet.py
from pydantic import BaseModel


class DepositAddressRequest(BaseModel):
    timeout: int


class DepositAddressResponse(BaseModel):
    operation_id: str
    address: str


class AwaitDepositRequest(BaseModel):
    operation_id: str


class AwaitDepositResponse(BaseModel):
    status: str


class WithdrawRequest(BaseModel):
    address: str
    amount_msat: str


class WithdrawResponse(BaseModel):
    txid: str
    fees_sat: int

File: wrappers/fedimint-ts/types/modules/wallet.ts
interface DepositAddressRequest {
    timeout: number;
}

interface DepositAddressResponse {
    operationId: string;
    address: string;
}

interface AwaitDepositRequest {
    operationId: string;
}

interface AwaitDepositResponse {
    status: string;
}

interface WithdrawRequest {
    address: string;
    amountMsat: number | 'all';
}

interface WithdrawResponse {
    txid: string;
    feesSat: number;
}

export type {
    DepositAddressRequest,
    DepositAddressResponse,
    AwaitDepositRequest,
    AwaitDepositResponse,
    WithdrawRequest,
    WithdrawResponse,
}
